packages/playground/blueprints/src/lib/steps/activate-plugin.ts

```diff
@@ -65,22 +64,61 @@ export const activatePlugin: StepHandler<ActivatePluginStep> = async (
 				}
 			}
 
-			if ( null === $response ) {
-				die('Plugin activated successfully');
-			} else if ( is_wp_error( $response ) ) {
-				throw new Exception( $response->get_error_message() );
+			if (is_wp_error($response)) {
+				die($response->get_error_message());
 			}
+		`,
+	});
+
+	/**
+	 * Instead of checking the plugin activation response,
+	 * check if the plugin is active by looking at the active plugins list.
+	 *
+	 * We have to split the activation and the check into two PHP runs
+	 * because some plugins might redirect during activation,
+	 * which would prevent any output that happens after activation from being returned.
+	 *
+	 * Relying on the plugin activation response is not reliable because if the plugin activation
+	 * produces any output, it will be threaded as an error.
+	 * See WordPress source code for more details:
+	 * https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-admin/includes/plugin.php#L733
+	 *
+	 * Because some plugins can create an output, we need to use output buffering
+	 * to ensure the 'true' response is not polluted by other outputs.
+	 * If the plugin activation fails, we will return the buffered output as it might
+	 * contain more information about the failure.
+	 */
+	const isActiveCheckResult = await playground.run({
+		code: `<?php
+			ob_start();
+			require_once( ${phpVar(docroot)}. "/wp-load.php" );
 
-			throw new Exception( 'Unable to activate plugin' );
+			$relative_plugin_path = str_replace( ${phpVar(
```

Comment:
Fun fact: we can use the `env: {}` option of `playground.run()` to pass input data and the `getenv()` function to read it.

Reply:
Also, this `str_replace` will replace all the occurences of `"$docroot/wp-content/plugins"` instead of removing bytes from the beginning of the path. `substr()` would not have this problem.

Reply:
Done in [cf2a580](https://github.com/WordPress/wordpress-playground/pull/2066/commits/cf2a5800b8bd65b88919703423b245cd17a4e080)